server/src/test/java/org/elasticsearch/search/fetch/FetchPhaseTests.java

```diff
@@ -1,31 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0 and the Server Side Public License, v 1; you may not use this file except
- * in compliance with, at your election, the Elastic License 2.0 or the Server
- * Side Public License, v 1.
- */
-
-package org.elasticsearch.search.fetch;
-
-import org.elasticsearch.test.ESTestCase;
-
-public class FetchPhaseTests extends ESTestCase {
-    public void testSequentialDocs() {
-        FetchPhase.DocIdToIndex[] docs = new FetchPhase.DocIdToIndex[10];
-        int start = randomIntBetween(0, Short.MAX_VALUE);
-        for (int i = 0; i < 10; i++) {
-            docs[i] = new FetchPhase.DocIdToIndex(start, i);
-            ++start;
-        }
-        assertTrue(FetchPhase.hasSequentialDocs(docs));
-
-        int from = randomIntBetween(0, 9);
-        start = docs[from].docId;
-        for (int i = from; i < 10; i++) {
-            start += randomIntBetween(2, 10);
-            docs[i] = new FetchPhase.DocIdToIndex(start, i);
-        }
-        assertFalse(FetchPhase.hasSequentialDocs(docs));
-    }
-}
```

Comment:
Do we get this test some place else?

I wonder if you could write some tests for `FetchPhaseDocsIterator` - like that it groups the doc ids correctly and stuff. It's a think I wanted to do when I built the first synthetic source iteration but couldn't because it was all tied into FetchPhase. But now that you've pulled it out it's possible!

Reply:
It's entirely unused, the actual logic is in StoredFieldLoader these days.

I'll add some tests to make sure this behaves as expected.